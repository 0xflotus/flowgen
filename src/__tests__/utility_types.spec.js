// @flow

import { compiler, beautify } from "..";

it("should handle utility types", () => {
  const ts = `
type A = Readonly<{a: number}>
type B = Partial<{a: number}>
type C = NonNullable<string | null>
type D = ReadonlyArray<string>
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
