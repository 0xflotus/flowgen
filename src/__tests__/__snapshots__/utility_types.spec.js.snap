// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle utility types 1`] = `
"declare type A = $ReadOnly<{
  a: number
}>;
declare type B = $Shape<{
  a: number
}>;
declare type C = $NonMaybeType<string | null>;
declare type D = $ReadOnlyArray<string>;
"
`;
